server/config/configresource/config_resource_test.go

```diff
@@ -0,0 +1,63 @@
+package configresource_test
+
+import (
+	"context"
+	"testing"
+
+	"github.com/gorilla/mux"
+	"github.com/kubeshop/tracetest/server/config/configresource"
+	"github.com/kubeshop/tracetest/server/id"
+	"github.com/kubeshop/tracetest/server/resourcemanager"
+	rmtests "github.com/kubeshop/tracetest/server/resourcemanager/testutil"
+	"github.com/kubeshop/tracetest/server/testmock"
+)
+
+func TestConfigResource(t *testing.T) {
+
+	db := testmock.MustGetRawTestingDatabase()
+	sampleConfig := configresource.Config{
+		ID:               "1",
+		Name:             "test",
+		AnalyticsEnabled: true,
+	}
+
+	rmtests.TestResourceType(t, rmtests.ResourceTypeTest{
+		ResourceType: "Config",
+		RegisterManagerFn: func(router *mux.Router) any {
+			db := testmock.MustCreateRandomMigratedDatabase(db)
+			configRepo := configresource.Repository(db)
+
+			manager := resourcemanager.New[configresource.Config]("Config", configRepo, id.GenerateID)
+			manager.RegisterRoutes(router)
+
+			return configRepo
+		},
+		Prepare: func(t *testing.T, op rmtests.Operation, bridge any) {
+			configRepo := bridge.(resourcemanager.ResourceHandler[configresource.Config])
+			switch op {
+			case rmtests.OperationGetSuccess,
+				rmtests.OperationUpdateSuccess,
+				rmtests.OperationDeleteSuccess,
+				rmtests.OperationListSuccess:
+				configRepo.Create(context.TODO(), sampleConfig)
```

Comment:
In this example, we create a new example config for the operations that require a preexisting config in the DB. This approach allows to not depend on the order of the operations